docs/en/framework/scopes.md

```diff
@@ -1,50 +1,208 @@
 # Framework — IoC Scopes
 
 An essential aspect of developing long-living applications is proper context management. In demonized applications,
-you are no longer allowed to treat user requests as a global singleton object and store references to its instance in 
+you are no longer allowed to treat user requests as a global singleton object and store references to its instance in
 your services.
 
-Practically it means that you must explicitly request context while processing user input. Spiral simplifies such 
-requests by using a global IoC container as a context carrier which allows you to call request specific instances as 
-global objects via context bounded scopes.
+Practically it means that you must explicitly request context while processing user input. Spiral offers a simple way to
+manage this by using a global IoC (Inversion of Control) container. This acts as a context carrier that allows you to
+request specific instances, scoped to a particular context, as if they were global objects.
 
-## Explanation
+## How It Works
 
-While processing some user request, the context-specific data is located in the IoC scope, available in the container
-only for a limited time. Such operation is performed using `Spiral\Core\ScopeInterface`->`runScope` method. Default
-spiral container `Spiral\Core\Container` implements this interface.
+When processing a user request, the relevant data for that request is placed in a scoped area of the IoC container. This
+data is only available for a limited time while processing that request.
+
+You can use the `Spiral\Core\ScopeInterface->runScope` method to achieve this. The default Spiral
+container, `Spiral\Core\Container`, implements this interface.
+
+**Here is an example**
 
 ```php
 use Psr\Container\ContainerInterface;
 
-$container->runScope(
-    [
-        UserContext::class => $user
-    ],
-    function (ContainerInterface $container) {
-    
+$container->runScoped(
+    closure: function (ContainerInterface $container) {
         dump($container->get(UserContext::class);
-
-    }
+    },
+    bindings: [UserContext::class => $user,],
 );
 ```
 
-The first argument of the `runScope` method is an array of values that should be bound to the container within the 
+The first argument of the `runScope` method is an array of values that should be bound to the container within the
 callback's scope.
 
-The callback function receives the container as its argument and can use the container to retrieve the values that were 
-bound to it within the callback's scope. This can be useful for creating a temporary scope within which certain values 
+The callback function receives the container as its argument and can use the container to retrieve the values that were
+bound to it within the callback's scope. This can be useful for creating a temporary scope within which certain values
```

Comment:
The closure will be autowired. There any parameters can be. The ContainerInterface will resolved as a container of the new scope.